migrations/sql/V2020.05.04_1059__OrderlyWebReportVersion.sql

```diff
@@ -0,0 +1,17 @@
+CREATE TABLE "orderlyweb_report_version" (
+"id"  TEXT NOT NULL,
+"published" BOOLEAN NOT NULL,
+FOREIGN KEY ("id") REFERENCES "report_version" ("id"),
+PRIMARY KEY("id")
+);
+
+INSERT INTO "orderlyweb_report_version"
+SELECT id, published
+FROM report_version;
+
+CREATE TRIGGER insert_orderlyweb_report_version
```

Comment:
I wonder about if there will be other things we do this to? We hold tags without needing to to a trigger already, so I don't see why we need to do this via a trigger necessarily.  So for tags we have:

```
CREATE TABLE "orderlyweb_report_tag" (
"report"  TEXT NOT NULL,
"tag" TEXT NOT NULL,
FOREIGN KEY ("report") REFERENCES "report" ("name"),
PRIMARY KEY("report", "tag")
);

CREATE TABLE "orderlyweb_report_version_tag" (
"report_version"  TEXT NOT NULL,
"tag" TEXT NOT NULL,
FOREIGN KEY ("report_version") REFERENCES "report_version" ("id"),
PRIMARY KEY("report_version", "tag")
);
```

would the same approach not work here? We could hold a table  `orderlyweb_report_published` which has just a reference to `report_version.id` (or possibly also holding something like the time at which the status changed).  If we're doing this with a trigger, do we want to generalise this table to be some sort of mirror of orderly's report list, ready to cope with the next bit of naturally 1:1 information?

There's a related issue of report deletion that I've written down in [VIMC-3844](https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-3844)

Reply:
Well tags isn't a 1:1 relationship, you might not add any tags for a report version at all. If we didn't have this trigger, we could manage it through code but it would be a bit fiddly - whenever you went to get or set the published state you'd have to deal with it maybe not being there and managing the insert when required. 

> If we're doing this with a trigger, do we want to generalise this table to be some sort of mirror of orderly's report list, ready to cope with the next bit of naturally 1:1 information?

Yeah, that's the idea of this table, that it should be all orderly web's managed data about a report version. If we need to add columns later, I guess we do that in a subsequent migration, and maybe also have to drop this trigger and create a new one.